source/base/parameter_handler.cc

```diff
@@ -3102,16 +3051,29 @@ MultipleParameterLoop::~MultipleParameterLoop ()
 
 
 
-bool MultipleParameterLoop::read_input (std::istream &input,
-                                        const std::string &filename,
-                                        const std::string &last_line)
+bool
+MultipleParameterLoop::read_input (std::istream &input,
+                                   const std::string &filename,
+                                   const std::string &last_line)
+{
+  MultipleParameterLoop::parse_input(input, filename, last_line);
+  return true;
+}
+
+
+
+void
+MultipleParameterLoop::parse_input (std::istream &input,
+                                    const std::string &filename,
+                                    const std::string &last_line)
 {
   AssertThrow (input, ExcIO());
 
-  bool x = ParameterHandler::read_input (input, filename, last_line);
-  if (x)
-    init_branches ();
-  return x;
+  // Note that, to avoid a stack overflow, we have to explicitly call the
+  // function that does the work and *not* a wrapper (which may be virtual and
+  // lead us back here)
+  ParameterHandler::parse_input (input, filename, last_line);
```

Comment:
No, that comment is new. You are correct (the term 'infinite recursion' is more accurate).